aibolit/patterns/send_null/send_null.py

```diff
@@ -9,59 +9,28 @@ class SendNull:
-                                    return position
-                else:
-                    for j in i:
-                        position = self.__find_position_recursively(j)
-
-        return position
-
-    # flake8: noqa
-    # after fix addition, errors are shown
     def value(self, filename: str) -> List[int]:
 
-        lst: List[int] = []
-        tree = build_ast(filename)
-
-        invocation_tree = tree.filter(javalang.tree.Invocation)
-        arg_list = [x for _, x in invocation_tree]
-
-        for argument in arg_list:
-            ternary_list = argument.filter(javalang.tree.TernaryExpression)
-            for _, expr in ternary_list:
-                if isinstance(expr.if_false, javalang.tree.Literal) and expr.if_false.value == 'null':
-                    if hasattr(argument, '_position'):
-                        lst.append(argument._position.line)
-                    else:
-                        position = self.__find_position_recursively(expr)
-                        lst.append(position)
-                if isinstance(expr.if_true, javalang.tree.Literal) and expr.if_true.value == 'null':
-                    if hasattr(argument, '_position'):
-                        lst.append(argument._position.line)
-                    else:
-                        position = self.__find_position_recursively(expr)
-                        lst.append(position)
-
-        for _, node in tree:
-            try:
-                for argument in node.arguments:
-                    if isinstance(argument, javalang.tree.Literal) and argument.value == "null" and \
-                            argument._position.line not in lst:
-                        lst.append(argument._position.line)
-            except (AttributeError, TypeError):
-                pass
-        lst = sorted(lst)
+        lines = set()
+        tree = AST.build_from_javalang(build_ast(filename))
+        invocatios_types = [
+            ASTNodeType.METHOD_INVOCATION,
+            ASTNodeType.EXPLICIT_CONSTRUCTOR_INVOCATION,
+            ASTNodeType.CLASS_CREATOR
+        ]
+        for node in tree.get_proxy_nodes(*invocatios_types):
+            for argument in node.arguments:
+                if (argument.node_type == ASTNodeType.LITERAL) and (argument.value == "null"):
+                    lines.add(argument.line)
+
+        for node in tree.get_proxy_nodes(ASTNodeType.TERNARY_EXPRESSION):
+            has_ternary_true_value = hasattr(node.if_true, 'value')
+            is_ternary_true_str = isinstance(node.if_true, str)
+            has_ternary_false_value = hasattr(node.if_false, 'value')
+            is_ternary_false_str = isinstance(node.if_false, str)
+            if (has_ternary_true_value and is_ternary_true_str and node.if_true.value == 'null') or \
+                    (has_ternary_false_value and is_ternary_false_str and node.if_false.value == 'null'):
+                lines.add(node.line)
+
+        lst = sorted(lines)
```

Comment:
I think all lines will come up already sorted. So maybe use list for lines?

Reply:
no, set are unordered:

` a_set = {6,7,8,9}`
`{8, 9, 6, 7}`

Reply:
I mean if we will use `list` for storing line number, they will probably come up all in right order. No need for `set` or `sorted`.